packages/editor/src/components/inserter/menu.js

```diff
@@ -198,11 +198,16 @@ export class InserterMenu extends Component {
 		if ( filterValue !== this.state.filterValue ) {
 			if ( ! filterValue ) {
 				openPanels = [ 'suggested' ];
-			} else if ( reusableItems.length ) {
-				openPanels = [ 'reusable' ];
-			} else if ( filteredItems.length ) {
-				const firstCategory = find( getCategories(), ( { slug } ) => itemsPerCategory[ slug ] && itemsPerCategory[ slug ].length );
-				openPanels = [ firstCategory.slug ];
+			} else {
+				openPanels = [];
+				if ( reusableItems.length ) {
+					openPanels.push( 'reusable' );
+				}
+				if ( filteredItems.length ) {
+					openPanels = openPanels.concat(
+						Object.keys( itemsPerCategory )
+					);
+				}
```

Comment:
I think it might be worthwhile to break this out to it's own helper to help read what's happening here. For example:

```jsx
filterOpenPanels: ( filterValue, itemsPerCategory ) => {
    if( filterValue === this.state.filterValue ) {
        return this.state.openPanels;
    }
    if ( ! filterValue ) {
        return [ 'suggested' ];
    }
    if ( reusableItems.length > 0 ) {
        return [ 'reusable' ];
    }
    if( filteredItems.length > 0) {
        return [ ...Object.keys( itemsPerCategory ) ]
    }
    return [];
}
// in filter...
this.setState( {
   openPanels: this.filterOpenPanels( filterValue, itemsPerCategory ), 
   //...
} );
```